Documentation improvements (#32)
launchdarkly/git-flag-parser: README.md

The command line program may be run manually, and executed in an environment of your choosing. Downloads for the latest release can be found [here](https://github.com/launchdarkly/git-flag-parser/releases/latest). Additionally, a docker image containing the git flag parser is available on the docker registry as [`ldactions/git-flag-parser`](https://hub.docker.com/r/ldactions/git-flag-parser)

macOS users should download the darwin release for their respective system architecture.

> The `git-flag-parser` requires [The Silver Searcher](https://github.com/ggreer/the_silver_searcher#installing) to be installed as a dependency, so make sure it has been installed and added to your system PATH before running the git-flag-parser.

Reviewer: super small nit, I'd prefer either "system's `$PATH`" or "system path"

A number of command-line arguments are available to the parser, some optional, and some required. Command line arguments may be passed to the program in any order.

### Required arguments

| Option | Description |
|-|-|

> | `accessToken` | LaunchDarkly [personal access token](https://docs.launchdarkly.com/docs/api-access-tokens) with writer-level access, or access to the `code-references` [custom role](https://docs.launchdarkly.com/v2.0/docs/custom-roles) resource |

Reviewer: we should have a ticket to update https://docs.launchdarkly.com/v2.0/docs/custom-roles with the code refs resource identifier.

Author: https://app.clubhouse.io/launchdarkly/story/29783/update-custom-role-docs

> | `dir` | Path to existing checkout of the git repo. If a cloneEndpoint is provided, this option is not required. |

Reviewer: ```suggestion
| `dir` | **Absolute** path to existing checkout of the git repo. Relative paths are not currently supported. |
```

| `projKey` | A LaunchDarkly project key. |
| `repoName` | Git repo name. Will be displayed in LaunchDarkly |

Here's an example shell invocation of the git-flag-parser for one of LaunchDarkly's demo repositories:, with the binary located in the current directory, and a minimal configuration:

```shell
./git-flag-parser \
  -accessToken="$YOUR_LAUNCHDARKLY_ACCESS_TOKEN" \

>   -dir="path/to/git/repo" \

Reviewer: ```suggestion
  -dir="/path/to/git/repo" \
```

  -repoName="my-repository" \
  -projKey="default"
```

### Optional arguments

| Option | Description | Default |
|-|-|-|
| `baseUri` | Set the base URL of the LaunchDarkly server for this configuration. Only necessary if using a private instance of LaunchDarkly. | `https://app.launchdarkly.com` |

> | `cloneEndpoint` | If provided, will clone the repo from this endpoint to the provided `dir`. If authentication is required, this endpoint should be authenticated. Supports the https protocol for git cloning. Example: `https://username:password@github.com/username/repository.git` | n/a |

Reviewer: Might as well remove this now right?

Author: Remove with https://github.com/launchdarkly/git-flag-parser/pull/33

| `contextLines` | The number of context lines to send to LaunchDarkly. If < 0, no source code will be sent to LaunchDarkly. If 0, only the line containing flag references will be sent. If > 0, will send that number of context lines above and below the flag reference. A maximum of 5 context lines may be provided. | -1 |
| `defaultBranch` | The git default branch. The LaunchDarkly UI will default to display code references for this branch. | "master" |

> | `exclude` | A regular expression defining the files and directories which the flag parser should exclude. | "" |

Reviewer: I think we should offer some examples and a regex specification. I don't think I even know how to use this correctly, e.g. which if the following is correct: `vendor/`, `vendor/*`, `vendor/.*`?

| `updateSequenceId` | An integer representing the order number of code reference updates. Used to version updates across concurrent executions of the flag parser. If not provided, data will always be updated. If provided, data will only be updated if the existing `updateSequenceId` is less than the new `updateSequenceId`. Examples: the time a `git push` was initiated, CI build number, the current unix timestamp. | n/a |

> | `repoHead` | The HEAD or ref to retrieve code references from. Should be provided if the `git push` was initiated on a non-master branch. | "master" | no |

Reviewer: the wording is a bit confusing here. also, can you remind me why we have this option?

Author: We have this option so we can decide which branch to checkout to before scanning for code references. I have a separate ticket filed for deciding if we should remove this or not.

> | `repoType` | The repo service provider. Used to correctly categorize repositories in the LaunchDarkly UI. Acceptable values: github\|bitbucket\|custom | "custom" |

Reviewer: i wonder if we should rename "custom" to "other." I remember someone during secret shopper not understanding the difference, and wondering if they should just use "custom." 

"correctly categorize repositories in the LaunchDarkly UI" is hard to understand. an alternative would be "Used to generate repository links in the LaunchDarkly UI.

| `repoUrl` | The display url for the repository. If provided for a github or bitbucket repository, LaunchDarkly will attempt to automatically generate source code links. Example: `https://github.com/launchdarkly/git-flag-parser` | "" |

> | `commitUrlTemplate` | If provided, LaunchDarkly will attempt to generate links to your Git service provider per commit. Example: `https://github.com/launchdarkly/git-flag-parser/commit/${sha}`. Allowed template variables: `branchName`, `sha`. If `commitUrlTemplate` is not provided, but `repoUrl` is provided, LaunchDarkly will automatically generate links for github or bitbucket repo types. | "" |

Reviewer: > If `commitUrlTemplate` is not provided, but `repoUrl` is provided, LaunchDarkly will automatically generate links for github or bitbucket repo types.

Do we actually do this? I would expect that we'd only do it if we get `repoUrl` _and_ `repoType`

Author: Yes, that's right. The end of that quote says "links for github or bitbucket repo types". I'll update it to be clearer.

> | `hunkUrlTemplate` | If provided, LaunchDarkly will attempt to generate links to your Git service provider per code reference. Example: `https://github.com/launchdarkly/git-flag-parser/blob/${sha}/${filePath}#L${lineNumber}`. Allowed template variables: `sha`, `filePath`, `lineNumber`. If `hunkUrlTemplate` is not provided, but `repoUrl` is provided, LaunchDarkly will automatically generate links for github or bitbucket repo types. | "" |

Reviewer: same question as above

```shell
./git-flag-parser \
  -accessToken="$YOUR_LAUNCHDARKLY_ACCESS_TOKEN" \

>   -dir="path/to/git/repo" \

Reviewer: ```suggestion
  -dir="/path/to/git/repo" \
```
